packages/ec-core/src/operator/selector/lexicase.rs

```diff
@@ -95,3 +100,200 @@ where
+    }
+
+    #[test]
+    fn multiple_copies_best() {
+        // The two 9s are the possible winners
+        let population = population_from_single_scores([5, 8, 9, 6, 3, 2, 0, 9]);
+
+        let lexicase = Lexicase::new(1);
+        let mut rng = rand::thread_rng();
+
+        let selected = *lexicase.select(&population, &mut rng).unwrap().genome();
+        assert!(
+            selected == 2usize || selected == 7usize,
+            "genome {selected} should have been 2 or 7"
+        );
+    }
+
+    #[test]
+    fn single_best_multiple_errors() {
+        // [9, 8] is the sole winner
+        let population =
+            population_from_scores([[5, 3], [8, 2], [9, 8], [6, 2], [3, 8], [2, 8], [0, 6]]);
+
+        let lexicase = Lexicase::new(2);
+        let mut rng = rand::thread_rng();
+
+        assert_eq!(&2, lexicase.select(&population, &mut rng).unwrap().genome());
+    }
+
+    #[test]
+    fn multiple_best_multiple_errors() {
+        let population = population_from_scores([
+            [5, 3],
+            [8, 2],
+            [9, 8], // A possible winner
+            [6, 2],
+            [3, 8],
+            [2, 8],
+            [0, 6],
+            [7, 9], // A possible winner
+        ]);
+
+        let lexicase = Lexicase::new(2);
+        let mut rng = rand::thread_rng();
+
+        let selected = *lexicase.select(&population, &mut rng).unwrap().genome();
+        assert!(
+            selected == 2usize || selected == 7usize,
+            "genome {selected} should have been 2 or 7"
+        );
+    }
+
+    // A name for this type to simplify things in the `proptest` test
+    // below.
+    type TestIndividual = EcIndividual<String, TestResults<u16>>;
+
+    // This test uses `proptest` to generate a random set of between 1 and 20
+    // vectors of two scores, and then converts those into a vector of
+    // `EcIndividual`s (i.e., a population). I then determine the largest of the
+    // first values (i.e., the highest score on the first test case), and then the
+    // largest of the second values. I make a new set of results that is those
+    // highest values with one (randomly chosen by proptest) incremented by 1.
+    // This new result is guaranteed to be "better" than any other in the list,
+    // so it should be what is selected after I add it to the population.
+    //
+    // There are several helper functions that exist just to support the somewhat
+    // complex logic of this test.
+    #[proptest]
+    fn selects_sole_best(
+        #[strategy(1..20usize)] pop_size: usize,
+        #[strategy(make_pop(#pop_size))] mut population: Vec<TestIndividual>,
+        #[strategy(0..1usize)] score_to_increase: usize,
+    ) {
+        prop_assert!(pop_size > 0);
+        prop_assert_eq!(pop_size, population.len());
+
+        let (winning_label, winning_individual) =
+            make_winning_individual(&population, score_to_increase);
+
+        population.push(winning_individual);
```

Comment:
We could probably move this into a strategy since it's kind of a input but this is totally fine as well :+1:

Reply:
I'll leave this for now, but the new issue #240 reminds us to come back to this.